### PR TITLE
dev-libs/openssl: update to 3.0.8

### PR DESCRIPTION
**Pull request by dongsupark**
Update `dev-libs/openssl` to 3.0.8, mainly to address:
* [CVE-2022-4203](https://nvd.nist.gov/vuln/detail/CVE-2022-4203)
* [CVE-2022-4304](https://nvd.nist.gov/vuln/detail/CVE-2022-4304)
* [CVE-2022-4450](https://nvd.nist.gov/vuln/detail/CVE-2022-4450)
* [CVE-2023-0215](https://nvd.nist.gov/vuln/detail/CVE-2023-0215)
* [CVE-2023-0216](https://nvd.nist.gov/vuln/detail/CVE-2023-0216)
* [CVE-2023-0217](https://nvd.nist.gov/vuln/detail/CVE-2023-0217)
* [CVE-2023-0286](https://nvd.nist.gov/vuln/detail/CVE-2023-0286)
* [CVE-2023-0401](https://nvd.nist.gov/vuln/detail/CVE-2023-0401)

Based on Gentoo commit [d06970048f80](https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=d06970048f80989bd42e225521661e2469f54a32).

Apply Flatcar patch based on commit [e0de828e38d5](https://github.com/flatcar/coreos-overlay/commit/e0de828e38d524895d7db308ec9170ff4e03d71a).

## Testing done

CI: http://jenkins.infra.kinvolk.io:8080/job/container/job/packages_all_arches/1252/cldsv/

- [x] Changelog entries added in the respective `changelog/` directory (user-facing change, bug fix, security fix, update)
- [x] Inspected CI output for image differences: `/boot` and `/usr` size, packages, list files for any missing binaries, kernel modules, config files, kernel modules, etc.

### CODE REVIEW

**Comment by dongsupark** [comment]
CI passed

**Review by JAORMX — APPROVED** [review]

File: dev-libs/openssl/openssl-3.0.8.ebuild [hunk 1]
```diff
@@ -19,7 +20,7 @@ else
 	SRC_URI="mirror://openssl/source/${MY_P}.tar.gz
 		verify-sig? ( mirror://openssl/source/${MY_P}.tar.gz.asc )"
 	#KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86 ~x86-linux"
-	KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~loong ~m68k ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86"
+	KEYWORDS="~alpha amd64 ~arm arm64 ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86 ~x86-linux"
```
Comment (krnowak) [inline dev-libs/openssl/openssl-3.0.8.ebuild:23]: Could we minimize the diff here (so don't delete ~loong and don't add ~x86-linux)?

**Review by krnowak — APPROVED** [review]
Just one nitpick, otherwise looks good, thanks.

File: dev-libs/openssl/openssl-3.0.8.ebuild (hunk 1 above)
Comment (dongsupark) [reply dev-libs/openssl/openssl-3.0.8.ebuild]: Done. Thanks.

**Pull request merged**